src/i2c.rs

```diff
@@ -21,7 +21,13 @@ crate::embedded_hal_error!(
 
 #[cfg(any(esp32, esp32s2))]
 const APB_TICK_PERIOD_NS: u32 = 1_000_000_000 / 80_000_000;
-#[cfg(not(any(esp32, esp32s2)))]
+
```

Comment:
Strange. Shouldn't this driver be very similar to `TimeDriver` in that it should support the PLL clocks on anything but the esp32?

Reply:
Also below, it seems the XTAL clock is used for non-esp32 chips. This does not sound right. As discussed on Matrix, the XTAL crystal might just not be there, so we agreed that for all MCUs, we should use (by default or hard-coded) a clock which is always available on the concrete MCU.

Reply:
In the TImerDriver we already had the mechanism in place to switch out the underlying clock from apb/pll to xtal. This is not the case in this driver. We always use the default one for i2c. The story for clocks and to that extend default clocks on i2c is different. For example for the esp32c2 it supports the RC_FAST and the XTAL clock but by default its using the XTAL one if you look [here](https://github.com/espressif/esp-idf/blob/0479494e7abe5aef71393fba2e184b3a78ea488f/components/soc/esp32c2/include/soc/clk_tree_defs.h#L235). Similar in the other cases the XTAL clock is used as far as i could see it. 

Our puplic api here is also not that nice - from a naming schema perspective - using the APBTickType name. Though i didn't touch it as my primary goal was to get it fixed, and both(i2c and Timer) fixes don't break the current public API. I think this can be changed to a better design when we use the new headers i2c_master.h and i2c_slave.h

Reply:
In practice every esp32 has an xtal if it wants to use Bluetooth or wifi as the internal oscillator is just not stable enough. But yeah technically you can run it without one. Also in low power modes you disable the xtal one. But well that is all besides the point that we don't expose a mechanism to switch it in this i2c driver and that it seams that the default one is xtal in upstream.

Reply:
> In practice every esp32 has an xtal if it wants to use Bluetooth or wifi as the internal oscillator is just not stable enough. But yeah technically you can run it without one. Also in low power modes you disable the xtal one. But well that is all besides the point that we don't expose a mechanism to switch it in this i2c driver and that it seams that the default one is xtal in upstream.

I see and I agree xtal is the default one for i2c.
So w.r.t. i2c, I agree we can merge as-is.

But then maybe I do not understand something in general, if you can help clarify:
On the esp32 SOC itself (as in, the "chip") - and I suppose it is the same for all other Espressif MCUs... there is actually no xtal?

So when you talk about in "practice", do you mean developer boards and professional custom PCBs? For PCBs I can imagine that to be the case, but for regular hobby boards, do they really have an xtal? I think I only have one with an xtal...? Yet, everybody uses Wifi and BT on these dev boards?

Are we even talking about the same thing, or am I mistaking "xtal" with something else (I'm thinking about the external 0.05c "crystal" which usually spins at 40mHz and you need to manually connect ot certain pins on the esp32)?

Reply:
In practice i mean the difference between an esp [module](https://www.espressif.com/en/products/modules) and an esp [chip](https://www.espressif.com/en/products/socs). Every module has an xtal. The modules are the one with the silver cap. You see some dev boards that only have the raw chip. And yes we are talking about the external crystal.